test/src/test/java/org/corfudb/integration/CorfuReplicationMultiSenderIT.java

```diff
@@ -0,0 +1,588 @@
+    private CorfuStore corfuStoreSink;
+
+    private final String sourceEndpoint1 = DEFAULT_HOST + ":" + sourceSiteCorfuPort1;
+
+    private final String sourceEndpoint2 = DEFAULT_HOST + ":" + sourceSiteCorfuPort2;
+    private final String sourceEndpoint3 = DEFAULT_HOST + ":" + sourceSiteCorfuPort3;
+    private final String sinkEndpoint = DEFAULT_HOST + ":" + sinkSiteCorfuPort;
+
+    private static final String TABLE_1 = "Table001";
+    private static final String TABLE_2 = "Table002";
+    private static final String TABLE_3 = "Table003";
+
+    private static final String TABLE_MERGED = "Table_Merged";
+    private static final String NAMESPACE = DefaultLogReplicationConfigAdapter.NAMESPACE;
+
+    private static final String STREAM_TAG = DefaultLogReplicationConfigAdapter.TAG_ONE;
+
+    private static final int THREE = 3;
+    private static final int NINE = 9;
+    private static final int FOUR = 4;
+    private static final int SIX = 6;
+    private static final int FIVE = 5;
+
+    private static final int TEN = 10;
+
+    public String pluginConfigFilePath;
+
+    private Table<Sample.StringKey, SampleSchema.ValueFieldTagOne,
+        Message> table1;
+    private Table<Sample.StringKey, SampleSchema.ValueFieldTagOne,
+        Message> table2;
+    private Table<Sample.StringKey, SampleSchema.ValueFieldTagOne,
+        Message> table3;
+    private Table<Sample.StringKey, SampleSchema.ValueFieldTagOne,
+        Message> mergedTable;
+
+    public CorfuReplicationMultiSenderIT(String pluginConfigFilePath) {
+        this.pluginConfigFilePath = pluginConfigFilePath;
+    }
+
+    // Static method that generates and returns test data
+    // (automatically test for two transport protocols: netty and GRPC)
+    @Parameterized.Parameters
+    public static Collection input() {
+
+        List<String> transportPlugins = Arrays.asList(
+            "src/test/resources/transport/grpcConfig.properties",
+            "src/test/resources/transport/nettyConfig.properties");
+
+        List<String> absolutePathPlugins = new ArrayList<>();
+        transportPlugins.forEach(plugin -> {
+            File f = new File(plugin);
+            absolutePathPlugins.add(f.getAbsolutePath());
+        });
+
+        return absolutePathPlugins;
+    }
+
+    @Test
+    public void testMergeTable() throws Exception {
+        // Setup Corfu on 3 LR Source Sites and 1 LR Sink Site
+        setupSourceAndSinkCorfu();
+
+        // Open maps on the Source and Destination Sites
+        openMaps();
+        log.debug("Write 3 records locally on each Source Cluster");
+
+        // Write data to all the source sites
+        writeData(corfuStoreSource1, TABLE_1, table1);
+        writeData(corfuStoreSource2, TABLE_2, table2);
+        writeData(corfuStoreSource3, TABLE_3, table3);
+
+        log.debug("Verify No Data Found on the Sink");
+        // Assert no data on the sink site
+        Assert.assertEquals(0, mergedTable.count());
```

Comment:
why don't we exercise the most basic case where each table receives updates independently? showing that we support that.. cause the merging is a high-level function but doesn't prove that everything arrived into the right target table separately necessarily. For all we know all updates could've been erroneously placed on table3 and still the merged table would succeed. I don't think we aim to test a merge listener but multi-source instead.

Reply:
makes sense.  Added a test for the same.